reference-implementation/lib/ReadableStream-impl.js

```diff
@@ -158,6 +160,57 @@ exports.implementation = class ReadableStreamImpl {
     aos.ReadableStreamReaderGenericRelease(reader);
     return promiseResolvedWith(undefined);
   }
+
+  static from(asyncIterable) {
+    let stream;
+    const iteratorRecord = GetIterator(asyncIterable, 'async');
+
+    const startAlgorithm = () => undefined;
+
+    function pullAlgorithm() {
+      let nextResult;
+      try {
+        nextResult = IteratorNext(iteratorRecord);
+      } catch (e) {
+        return promiseRejectedWith(e);
+      }
+      const nextPromise = promiseResolvedWith(nextResult);
+      return transformPromiseWith(nextPromise, iterResult => {
+        if (typeof iterResult !== 'object') {
```

Comment:
`null` should also throw. There might be a `typeIsObject` predicate in one of the helper files? Or maybe I killed it when we moved stuff to IDL and it became unnecessary.

(Tests would be ideal to catch this case.)